http/percent_test.go

```diff
@@ -0,0 +1,35 @@
+package http_test
+
+import (
+	"github.com/kkrull/gohttp/http"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PercentDecode", func() {
+	It("returns a string unchanged that has no percent triplets in it", func() {
+		Expect(http.PercentDecode("abcd")).To(Equal("abcd"))
+	})
+
+	It("decodes a % triplet into the ASCII character for its hexadecimal code", func() {
+		Expect(http.PercentDecode("%3C")).To(Equal("<"))
+	})
+
+	It("retains characters after a percent triplet", func() {
+		Expect(http.PercentDecode("%3Cabc")).To(Equal("<abc"))
+	})
+
+	It("decodes multiple percent triplets", func() {
+		Expect(http.PercentDecode("%3Ca%3Cb")).To(Equal("<a<b"))
+	})
+
+	It("returns an error when % is followed by no characters", func() {
+		_, err := http.PercentDecode("abc%")
+		Expect(err).To(MatchError("% followed by fewer than 2 characters: abc%"))
+	})
+
+	It("returns an error when % is followed by 1 character", func() {
+		_, err := http.PercentDecode("abc%1")
+		Expect(err).To(MatchError("% followed by fewer than 2 characters: abc%1"))
```

Comment:
I might consider a custom error you could match against so that you don't need to repeat the exact string formatting

Reply:
Done